src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -13,11 +13,18 @@
 using Microsoft.Identity.Client.Utils;
 using Microsoft.Win32.SafeHandles;
 using Microsoft.Identity.Client.Platforms.net45.Native;
+using System.Collections.Concurrent;
+using System.Threading;
 
 namespace Microsoft.Identity.Client.Platforms.net45
 {
     internal class NetDesktopCryptographyManager : ICryptographyManager
     {
+        private static ConcurrentDictionary<string, RSA> CertificateToRsaMap = new ConcurrentDictionary<string, RSA>();
+        private static ConcurrentDictionary<string, RSACryptoServiceProvider> CertificateToRsaCspMap = new ConcurrentDictionary<string, RSACryptoServiceProvider>();
+        private static int maximumCertificateToRsaMap = 1000;
```

Comment:
Why does one dictionary have a max size while another does not?

Reply:
Oh, they should both have max size.

Reply:
Updated. Resolved.